src/rhsmlib/services/register.py

```diff
@@ -44,6 +45,16 @@ def register(self, org, activation_keys=None, environment=None, force=None, name
         if kwargs:
             raise exceptions.ValidationError(_("Unknown arguments: %s") % kwargs.keys())
 
+        if no_insights is True:
+            try:
+                with open('/dev/null', 'w') as devnull:
+                    subprocess.call(['/usr/bin/systemctl', 'mask', '--now', 'insights-register.path'], stdout=devnull,
+                                    stderr=devnull)
+            except:
+                # ignore failures here in case we're running in a container, or some other issue prevents disabling
+                # insights auto-register
+                log.warn("Failed to disable automatic insights registration")
```

Comment:
`log.warn()` is considered as deprecated. Please change it to `log.warning()`.

Reply:
Done